CMakeLists.txt

```diff
@@ -28,6 +26,12 @@ option (SPEC_ARPA2
 	"Build inlucde files from ASN.1 specs for ARPA2 projects"
 	ON)
 
+set(Quick-DER_MAJOR_VERSION 0)
+set(Quick-DER_MINOR_VERSION 1)
+set(Quick-DER_PATCH_VERSION 5)  # RC5, will move to 0.2 for release
+set(Quick-DER_VERSION
+	${Quick-DER_MAJOR_VERSION}.${Quick-DER_MINOR_VERSION}.${Quick-DER_PATCH_VERSION})
+
```

Comment:
Zoals afgesproken ga ik niet onder je kleiduiven schieten, daarom post ik hier.
Dit patroon heb ik voor LillyDAP gebruikt, om tags [volgens ons versieschema](https://github.com/arpa2/howto/blob/master/release.md) te automatiseren:
```
#
# Version Information
#

find_program (GIT git)

exec_program (
                ${GIT}
                ARGS diff --quiet
                RETURN_VALUE GIT_HAVE_CHANGES
)

exec_program (
                ${GIT}
                ARGS describe --tags --match 'version-*.*-*'
                OUTPUT_VARIABLE GIT_VERSION_INFO
)

unset (GIT CACHE)

string (
        REGEX REPLACE "^version-([1-9][0-9]*|0)[.]([1-9][0-9]*|0)-(.*)$"
        "\\1"
        GIT_VERSION_MAJOR
        ${GIT_VERSION_INFO}
)

string (
        REGEX REPLACE "^version-([1-9][0-9]*|0)[.]([1-9][0-9]*|0)-(.*)$"
        "\\2"
        GIT_VERSION_MINOR
        ${GIT_VERSION_INFO}
)

if (GIT_HAVE_CHANGES EQUAL 0)

        string (
                REGEX REPLACE "^version-([1-9][0-9]*|0)[.]([1-9][0-9]*|0)-(.*)$"
                "\\3"
                GIT_VERSION_PATCHLEVEL
                ${GIT_VERSION_INFO}
        )

        set (
                USER_SUPPLIED_PATCHLEVEL
                "${GIT_VERSION_PATCHLEVEL}"
                CACHE STRING "User-override for patch level under ${GIT_VERSION_MAJOR}.${GIT_VERSION_MINOR}"
        )

else()

        exec_program (
                date
                ARGS '+%Y%m%d-%H%M%S'
                OUTPUT_VARIABLE GIT_CHANGES_TIMESTAMP
        )
        set (GIT_VERSION_PATCHLEVEL "local-${GIT_CHANGES_TIMESTAMP}")
        message (WARNING "Git reports local changes, fixing patch level to local-${GIT_CHANGES_TIMESTAMP}")

        unset (USER_SUPPLIED_PATCHLEVEL CACHE)

endif()
```
en dan later, bij het packagen,
```
set (CPACK_PACKAGE_VERSION_MAJOR ${GIT_VERSION_MAJOR})
set (CPACK_PACKAGE_VERSION_MINOR ${GIT_VERSION_MINOR})
if (GIT_HAVE_CHANGES EQUAL 0)
        set (CPACK_PACKAGE_VERSION_PATCH ${USER_SUPPLIED_PATCHLEVEL})
else()
        set (CPACK_PACKAGE_VERSION_PATCH "local-${GIT_CHANGES_TIMESTAMP}")
endif()
```
Denk je dat dit zinvol zou zijn?  (Uiteraard ga jij het in `.cmake` files stoppen, dat mechanisme heb ik nog niet helemaal in de smiezen qua zoekpaden en werking enzo.)
Ik snap inmiddels ook hoe zulke versies door kunnen dringen in je source code.  En ik begin te begrijpen waarom je hier extra een `${Quick-DER_xxxxx_VERSION}` definieert voor de CMake package repo.  Ik krijg weer de neiging om op een sigaar te kauwen... (en jou te zien als een zwaar met goud behangen geniale lasser).